### DISCUSSION TITLE
楷体 | /cn/2017/02/kaiti/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#862] [Comments]
https://yihui.org/cn/2017/02/kaiti/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
嗯，iOS 上默认没有楷体，Linux 和部分 Windows 10 用户也有同样的问题。

要想解决 Word 问题，得好好锻炼身体，争取比 Word 用户长命……

> ——原帖发布于 2017-02-14 02:48:03

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
输出格式用 bookdown::word_document2 可以作为近似解决方案，但有些细节之处无法做到像 HTML 和 LaTeX 一样高质量。我相信 Word 和 LaTeX 终归是要死掉的，唯有 HTML 会永生。

> ——原帖发布于 2017-02-14 21:39:01

**Reply by yihui** [reply]
对，就是这个问题。如果他们只是读文档，那没啥问题，就怕改，一改就完了，把那些改动再迁移回 Markdown 会死人的。

> ——原帖发布于 2017-02-15 15:27:17

**Reply by yihui** [reply]
深表同情……

> ——原帖发布于 2017-02-19 06:07:19

**Reply by yihui** [reply]
你可能有一个 _output.yml 文件吧，如果有的话，把 bookdown::word_document2: default 放到那里面去。然后就是看你怎么编译的了，如果是点 Build 按钮，要把它上面的下拉菜单展开，选中 bookdown::word_document2，否则它会记忆你上次选择的输出格式。

> ——原帖发布于 2017-02-23 19:09:19

**Reply by yihui** [reply]
哦，这里的 Knit 和 Build 两个按钮是容易让人糊涂。

> ——原帖发布于 2017-02-23 22:13:28

**Reply by yihui** [reply]
{#identifier .unnumbered}

重复文件名的问题应该来自 Pandoc，我确实对中文的标题的 ID 有特殊处理，因为中文标题无法生成 ID，所以我会自动把汉字编码为字母数字，自动添加 ID。但中英混合的标题我就没辙了，理论上也有可能解决，就像你说的自动添加数字编号。目前还是手工添加 #identifier 吧。

> ——原帖发布于 2017-02-26 03:32:11

**Reply by yihui** [reply]
哈哈，过奖。我从07年开始写我的第一本中文书《现代统计图形》时就觉得写书的感觉不对，作者不应该一直被格式问题烦扰，我那本书迟迟没有出版，部分原因也是由于我这些年绕了相当大一个圈子，如你说说，重造了 knitr 轮子取代 Sweave，再用 Markdown 替代 LaTeX，再稍微扩展一下 Markdown 语法，对一般的书籍来说，这一套工具链应该是在易用性和功能性中间达到一个很好的平衡了。不过直到现在，我仍然没有做好中文书的支持（主要是 LaTeX 模板部分），我大概知道怎么弄，只是要等下一段闲暇时间了，现在得主攻 blogdown，等 blogdown 尘埃落定，我的棋也就快下完了：R Markdown 写报告、论文、书、网站、幻灯片，我大致都有满意的答案了（虽然要改进的地方还不少）。

> ——原帖发布于 2017-02-27 03:59:18

**Reply by yihui** [reply]
是的，工具链有点长，主要是因为我们要取不同工具的精华部分。

> ——原帖发布于 2017-02-27 04:00:10

**Reply by yihui** [reply]
是的，要控制好重新发明轮子的程度，用 Markdown 完全替代 LaTeX 没有必要，不然语法肯定堕落成 LaTeX 了。在 bookdown 的开发过程中，我时刻需要打的一种仗就是说服用户放弃他们想要的功能。

云端是必须的。这个我早跟公司提过了，想法跟你一样，虽然基于 R Markdown 的工具相对简单，因为工具链太长，普通用户可能会被这个链条吓到，最好的方式就是我们在服务器上都配置好了，用户只要注册，进来点两下按钮就可以开始写书了。

学术期刊在我们的有生之年应该不会接受 Markdown 投稿，我的下一个项目就是要再发明一个轮子取代学术期刊，当然这个路会有点长，属于堂吉诃德战风车，但长远来看，我还是挺有信心的。

> ——原帖发布于 2017-02-27 17:50:19

**Reply by yihui** [reply]
这是一方面（还需要强调一下 bookdown 也可以像普通 R Markdown 一样生成单页面输出）；另一方面是 bookdown 扩充了 Markdown 的语法，参见 bookdown 书的第二章，普通的 R Markdown 不能给图片、表格编号并交叉引用，也不能写编号的公式、定理或证明，等等。

> ——原帖发布于 2017-02-28 19:07:00

**Reply by yihui** [reply]
深表同意，菜单确实有些乱，容易把人搞糊涂了，像 R Documentation 这一项几乎不会有人用，而 R Presentation 也容易让人以为是 R Markdown Presentation。

关于 bookdown 入口，像前面的云端服务一样，你又一次精确命中一个硬需求。其实这个入口我们已经做好了，只是目前还在开发版本中，你可以去 [dailies.rstudio.com](http://dailies.rstudio.com) 下载，然后菜单 File - New Project - New Directory，就能看到 bookdown 项目了。目前还有点粗糙，欢迎提供反馈意见。

> ——原帖发布于 2017-03-01 04:45:09

[a comment by an automated account is not shown]

**Comment by chuxinyuan** [comment]
我发现搜“字体”可以搜出一堆文章啊！哈哈哈……。

关于字体。我认为字体小的时候，眼睛首先要解决是能否看清楚的问题，所以小字无衬线的雅黑占优，衬线字体就显得“张牙舞爪”；字体大一点的时候，是否能看清楚这个问题不存在了，审美问题就凸显出来了，有棱有角的衬线字体，比如宋体、仿宋和楷体表现力就更好点，婀娜多姿，而衬线字体站在那里就很呆板了。
